docs/active_storage_migration.md

```diff
@@ -0,0 +1,387 @@
+rails> ds.where(as_converted_preview_document_checksum: nil).count
+# Should be zero
+
+# Number of documents with a valid checksum
+# A valid checksum ends with '==', such as '4+g5Nmjax5aCg3HtRmCO5Q=='
+rails> ds.select { |d| d.as_document_checksum&.match(/==$/) }.count
+# Should be the same as ds.count, above
+
+# Number of preview documents with a valid checksum
+# A valid checksum ends with '==', such as '4+g5Nmjax5aCg3HtRmCO5Q=='
+rails> ds.select { |d| d.as_converted_preview_document_checksum&.match(/==$/) }.count
+# Should be the same as ds.count, above
+```
+
+3) Migrate assets details from Paperclip to Active Storage:
+
+```bash
+$ bundle exec rails 'storage:migrate[documents]'
+```
+
+4) Confirm that all documents have been migrated:
+
+```ruby
+# List of Paperclip filenames for original documents
+rails> pc_document_filenames = Message.pluck(:document_file_name).sort
+
+# List of Active Storage filenames for original documents
+rails> as_document_filenames = ActiveStorage::Attachment.where(record_type: 'Document', name: 'document').map(&:filename).map(&:to_s).sort
+
+rails> pc_document_filenames == as_document_filenames
+# => true inducates that the migration is successful
+# => false indicates that the migraiton has failed
+
+# List of Paperclip filenames for preview documents
+rails> pc_preview_filenames = Message.pluck(:document_file_name).sort
+
+# List of Active Storage filenames for preview documents
+rails> as_preview_filenames = ActiveStorage::Attachment.where(record_type: 'Document', name: 'converted_preview_document').map(&:filename).map(&:to_s).sort
+
+rails> pc_preview_filenames == as_preview_filenames
+# => true inducates that the migration is successful
+# => false indicates that the migraiton has failed
+```
+
+5) Test that a sample document can be downloaded via Active Storage (this will only work when S3 is used for storage):
+
+```ruby
+# Get a download url for a sample document
+rails> ActiveStorage::Attachment.where(record_type: 'Document', name: 'document').sample.service_url
+# => "https://cloud-platform-..."
+# Paste this url into a browser and confirm that it can be downloaded
+
+# Get a download preview url for a sample document
+rails> ActiveStorage::Attachment.where(record_type: 'Document', name: 'converted_preview_document').sample.service_url(disposition: 'inline')
+# => "https://cloud-platform-..."
+# Paste this url into a browser and confirm that it can be previewed in the browser
+
+# Note that these url will expire in 5 minutes
+```
+
+6) (Final step) Activate Active Storage for documents by deploying PR #3739 (CBO-1693)
+
+### Testing
+
+1) Find an case with a message attachment. Confirm that evidence documents can
+   be downloaded and viewed in browser.
+2) As a caseworker, confirm that all documents can be downloaded as a zip file.
+   can be downloaded correctly.
+3) Add a new case and confirm that documents can be accessed as in steps (1)
+   and (2)
+
+### Rollback
```

Comment:
[comment as for stats report](https://github.com/ministryofjustice/Claim-for-Crown-Court-Defence/pull/3713/files#r581210705)